File: tests/examples/python_fmu/resources/adder.py
from fmi2 import FMU


class Adder(FMU):
    def __init__(self) -> None:

        self.real_a = 0.0
        self.real_b = 0.0

        self.integer_a = 0
        self.integer_b = 0

        self.boolean_a = False
        self.boolean_b = False

        self.string_a = ""
        self.string_b = ""

    @property
    def real_c(self):
        return self.real_a + self.real_b

    @property
    def integer_c(self):
        return self.integer_a + self.integer_b

    @property
    def boolean_c(self):
        return self.boolean_a and self.boolean_b

    @property
    def string_c(self):
        return self.string_a + self.string_b

File: tool/unifmu/resources/backends/java_fmu/src/main/java/fmu/Adder.java
package fmu;

public class Adder extends FMI2FMU {

    public double real_a;
    public double real_b;
    public double real_c;

    public int integer_a;
    public int integer_b;
    public int integer_c;

    public boolean boolean_a;
    public boolean boolean_b;
    public boolean boolean_c;

    public String string_a;
    public String string_b;
    public String string_c;

    public Adder() {
        super();

    }

    @Override
    public FMI2Status exitInitializationMode() {
        real_a = 0;
        real_b = 0;
        integer_a = 0;
        integer_b = 0;
        boolean_a = false;
        boolean_b = false;
        string_a = "";
        string_b = "";

        update_outputs();

        return FMI2Status.OK;
    }

    private void update_outputs() {
        real_c = real_a + real_b;
        integer_c = integer_a + integer_b;
        boolean_c = boolean_a || boolean_b;
        string_c = string_a + string_b;
    }

    @Override
    public FMI2Status doStep(double currentTime, double stepSize, boolean noStepPrior) {
        update_outputs();

        return FMI2Status.OK;
    }

    @Override
    public byte[] serialize() throws RuntimeException {
        // TODO Auto-generated method stub
        return super.serialize();
    }

    @Override
    public FMI2Status deserialize(byte[] state) throws RuntimeException {
        // TODO Auto-generated method stub
        return super.deserialize(state);
    }

}
